pkg/controller.v1/pytorch/pytorchjob_controller.go

```diff
@@ -465,9 +465,15 @@ func (r *PyTorchJobReconciler) UpdateJobStatusInApiServer(job interface{}, jobSt
 	return nil
 }
 
-// SetClusterSpec sets the cluster spec for the pod
+// SetClusterSpec sets the cluster spec and init container for the pod
 func (r *PyTorchJobReconciler) SetClusterSpec(job interface{}, podTemplate *corev1.PodTemplateSpec, rtype, index string) error {
-	return SetPodEnv(job, podTemplate, rtype, index)
+	if err := setPodEnv(job, podTemplate, rtype, index); err != nil {
+		return err
+	}
+	if err := setInitContainer(job, podTemplate, rtype, index, r.Log); err != nil {
```

Comment:
minor: I feel init container should be extracted as a util func. We can refactor to two steps
1. Create init container (input is the service to nslookup) -> common
2. Append the initContainer -> framework specific

Other framework may not have this requirement yet. Once they do, we do some refactor later.

Reply:
SGTM. Could we do it in another PR? I think it is kind of refactor, I can help improve it after the PR is merged.

/assign

Reply:
Let's have separate PR later and don't block this one